node/processingOnlyNode/coreComponents.go

```diff
@@ -0,0 +1,381 @@
+	apiEconomicsData              process.EconomicsDataHandler
+	ratingsData                   process.RatingsInfoHandler
+	rater                         sharding.PeerAccountListAndRatingHandler
+	genesisNodesSetup             sharding.GenesisNodesSetupHandler
+	nodesShuffler                 nodesCoordinator.NodesShuffler
+	epochNotifier                 process.EpochNotifier
+	enableRoundsHandler           process.EnableRoundsHandler
+	roundNotifier                 process.RoundNotifier
+	epochStartNotifierWithConfirm factory.EpochStartNotifierWithConfirm
+	chanStopNodeProcess           chan endProcess.ArgEndProcess
+	genesisTime                   time.Time
+	chainID                       string
+	minTransactionVersion         uint32
+	txVersionChecker              process.TxVersionCheckerHandler
+	encodedAddressLen             uint32
+	nodeTypeProvider              core.NodeTypeProviderHandler
+	wasmVMChangeLocker            common.Locker
+	processStatusHandler          common.ProcessStatusHandler
+	hardforkTriggerPubKey         []byte
+	enableEpochsHandler           common.EnableEpochsHandler
+}
+
+type ArgsCoreComponentsHolder struct {
+	Cfg                 config.Config
+	EnableEpochsConfig  config.EnableEpochs
+	RoundsConfig        config.RoundConfig
+	EconomicsConfig     config.EconomicsConfig
+	ChanStopNodeProcess chan endProcess.ArgEndProcess
+	NodesSetupPath      string
+	GasScheduleFilename string
+	NumShards           uint32
+	WorkingDir          string
+}
+
+func CreateCoreComponentsHolder(args ArgsCoreComponentsHolder) (factory.CoreComponentsHolder, error) {
+	var err error
+	instance := &coreComponentsHolder{}
+
+	instance.internalMarshaller, err = marshalFactory.NewMarshalizer(args.Cfg.Marshalizer.Type)
+	if err != nil {
+		return nil, err
+	}
+	instance.txMarshaller, err = marshalFactory.NewMarshalizer(args.Cfg.TxSignMarshalizer.Type)
+	if err != nil {
+		return nil, err
+	}
+	instance.vmMarshaller, err = marshalFactory.NewMarshalizer(args.Cfg.VmMarshalizer.Type)
+	if err != nil {
+		return nil, err
+	}
+	instance.hasher, err = hashingFactory.NewHasher(args.Cfg.Hasher.Type)
+	if err != nil {
+		return nil, err
+	}
+	instance.txSignHasher, err = hashingFactory.NewHasher(args.Cfg.TxSignHasher.Type)
+	if err != nil {
+		return nil, err
+	}
+	instance.uint64SliceConverter = uint64ByteSlice.NewBigEndianConverter()
+	instance.addressPubKeyConverter, err = factoryPubKey.NewPubkeyConverter(args.Cfg.AddressPubkeyConverter)
+	if err != nil {
+		return nil, err
+	}
+	instance.validatorPubKeyConverter, err = factoryPubKey.NewPubkeyConverter(args.Cfg.ValidatorPubkeyConverter)
+	if err != nil {
+		return nil, err
+	}
+
+	instance.pathHandler, err = storageFactory.CreatePathManager(
+		storageFactory.ArgCreatePathManager{
+			WorkingDir: args.WorkingDir,
+			ChainID:    args.Cfg.GeneralSettings.ChainID,
+		},
+	)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO check if we need the real watchdog
```

Comment:
can remove the TODO, the watchdog should be of type disabled

Reply:
done